### PR TITLE
Cleanup of ZM for conversion

### PR DESCRIPTION
**Pull request by cacraigucar**
This PR saves the separated routines and the CAM "use" statements have been removed.  This is the first step in the CCPP'ization of ZM.  The purpose of this PR is to begin having CAM use ZM from this repo (and hence needs an official tag).   Further refinement is expected.

Partially addresses #66

### CODE REVIEW

**Review by nusbaume — APPROVED** [review]
Looks good!  Left a bunch of comments, but mostly just so we can remember to deal with some additional issues and concerns later before finishing the full CCPP conversion needed for CAM-SIMA.

File: zm/zm_convr.F90
Comment (gold2718) [reply zm/zm_convr.F90:174]: Not yet but there could be? CAMDEN has the `physics_suite` variable (and associated function in the runtime object). This name is checked against the list in `ccpp_physics_suite_list` from the CCPP host cap.
I suggest adding a new standard name (with request to add it to the [dictionary](https://github.com/ESCOMP/CCPPStandardNames)), `ccpp_physics_suite`, which would contain this string. Then add it to a new host table (`phys_init`) in phys_comp.meta. That should make it available to the scheme (if it adds it as an input).
Thoughts?

**Comment by nusbaume** [comment]
@cacraigucar I just realized that I forgot to include in my review a reminder to add an atmospheric_physics `ChangeLog` entry before this is merged.  Thanks!

File: zm/zm_convr.F90
Comment (nusbaume) [reply zm/zm_convr.F90:174]: @gold2718 that sounds like a good general solution to me!

However, after thinking about this specific case some more I wonder if a possibly better solution for the ZM scheme is to just add a new namelist variable to turn on/off the non-dilute buoyancy used in CAM3.  

I feel doing it that way will make this particular scheme more portable (i.e. it won't be looking for a specific CAM-SIMA configuration).  We can then always set the default namelist value to be dependent on the physics suite one is using, which is a feature CAM-SIMA has already.

File: zm/zm_conv_convtran.F90
Comment (gold2718) [reply zm/zm_conv_convtran.F90:1]: FYI, this rule (module name conforms to the filename) is in the [Fortran Coding Standards](https://github.com/ESCOMP/CAM/wiki/CAM-Coding-Standards#fortran-coding-standards) under **SHOULD**.

**Comment by nusbaume** [comment]
@cacraigucar Just adding a request here to see if you could change the standard name `lagrangian_vertical_coordinate` in `utilities/geopotential_temp.meta` to `do_lagrangian_vertical_coordinate`?  I realized I had added the wrong standard name to this physics scheme in the last PR.  Thanks!

File: zm/zm_convr.F90
Comment (mwaxmonsky) [inline zm/zm_convr.F90:2978]: Not sure if its possible but if we could find someone that can document this section as well, that might be worthwhile.

File: zm/zm_convr.F90
Comment (mwaxmonsky) [inline zm/zm_convr.F90:2052]: Similar documentation comment here.

**Review by mwaxmonsky — APPROVED** [review]
Just one comment on a use statement in zm_convr.F90 and a few documentation points that are outside the scope of this PR but figured they'd be worth mentioning while we were looking at it but no reason to hold things up due to those.

Great job @cacraigucar, looks good to me!

File: zm/zm_convr.F90
Comment (nusbaume) [inline zm/zm_convr.F90:408]: Should we go ahead and remove the commented-out variables here and below?

File: zm/zm_convr.F90
Comment (nusbaume) [inline zm/zm_convr.F90:1610]: I am not sure if passing out `lchnk,icol` will work because there is no `lchnk` in CAM-SIMA.  Just an FYI to future you.

**Review by nusbaume — APPROVED** [review]
The new updates look good to me!  Just a couple of notes (which can be dealt with in the next round if preferred).

File: zm/zm_convr.F90
Comment (cacraigucar) [reply zm/zm_convr.F90]: deleted

File: zm/zm_convr.F90
Comment (cacraigucar) [reply zm/zm_convr.F90]: Noted this in my comment in the code

**Review by mwaxmonsky — COMMENTED** [review]
Great work @cacraigucar!  Just some commented out code that I think we can safely delete based on discussions with @nusbaume.

The other thing I was wondering if there's any value to is looking at the formatting?  There's several sections of code that change formatting ad-hoc and was wondering if we could make it consistent with the rest of the SIMA code base before we merge?  It might take a bit of working through and I'd be happy to help and I think it would be valuable to start with as clean of a slate as possible.

**Comment by cacraigucar** [comment]
@swrneale - Are you okay with the removal of all the commented out lines that @mwaxmonsky has indicated?

**Comment by mwaxmonsky** [comment]
@cacraigucar Thanks for taking care of the commented out code!

I was just wondering if there was any discussion on the formatting side of things?  Don't think that should hold up this PR and we can open a new one if it's worth pursuing but if it's reasonable, I'd be happy to help with that as well.

**Comment by cacraigucar** [comment]
> @cacraigucar Thanks for taking care of the commented out code!
> 
> I was just wondering if there was any discussion on the formatting side of things? Don't think that should hold up this PR and we can open a new one if it's worth pursuing but if it's reasonable, I'd be happy to help with that as well.

No there was no discussion on formatting.  I left it in the weird state that it is.  (I assume you are talking about the spaces in the calling lists, but I suppose there may be other locations as well).  At this point, if you'd like to make the formatting edits for the next time around, feel free.   I'd like this code to look as close to the CAM version so when @rneale brings in his changes, it is apparent to see where they go in relation to his personal fork.

**Comment by cacraigucar** [comment]
Also, I should point out that I am deferring removing the zm_org logic as well with this commit.

**Comment by swrneale** [comment]
Also I will work on finally expanding the org infrastructure as per this [pre-historic](https://github.com/ESCOMP/CAM/issues/190) issue over on ESCOMP

**Pull request merged**
